packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -78,6 +97,13 @@ class PlatformViewsService {
         if (_focusCallbacks.containsKey(id)) {
           _focusCallbacks[id]!();
         }
+      case 'navigatedOut':
+        final Map<dynamic, dynamic> args = call.arguments as Map<dynamic, dynamic>;
+        final int id = args['id'] as int;
+        final int reason = args['reason'] as int;
+        if (_loseFocusCallbacks.containsKey(id)) {
+          _loseFocusCallbacks[id]!(NavigationReason.values[reason]);
```

Comment:
This might break if someone reorders the enum values of `NavigationReason`, or, adds a new reason before an existing reason.

Some potential mitigations:

1. Make the `reason` argument a string value of `programmatic`, `forward`, `backward`
2. Manually decode integer values to a `NavigationReason` instead of using `NavigationReason.values`.